lib/validators/json-schema-legacy.js

```diff
@@ -0,0 +1,182 @@
+const amanda = require('amanda');
+const tv4 = require('tv4');
+const jsonPointer = require('json-pointer');
+
+const { JsonSchemaValidator, META_SCHEMA } = require('./json-schema-next');
+const { ValidationErrors } = require('./validation-errors');
+
+/**
+ * Returns a proper article for a given string.
+ * @param {string} str
+ * @returns {string}
+ */
+function getArticle(str) {
+  return ['a', 'e', 'i', 'o', 'u'].includes(str.toLowerCase()) ? 'an' : 'a';
+}
+
+const jsonSchemaOptions = {
+  singleError: false,
+  messages: {
+    minLength: (prop, val, validator) =>
+      `The ${prop} property must be at least ${validator} characters long (currently ${val.length} characters long).`,
+    maxLength: (prop, val, validator) =>
+      `The ${prop} property must not exceed ${validator} characters (currently${val.length} characters long).`,
+    length: (prop, val, validator) =>
+      `The ${prop} property must be exactly ${validator} characters long (currently ${val.length} characters long).`,
+    format: (prop, val, validator) =>
+      `The ${prop} property must be ${getArticle(
+        validator[0]
+      )} ${validator} (current value is ${JSON.stringify(val)}).`,
+    type: (prop, val, validator) =>
+      `The ${prop} property must be ${getArticle(
+        validator[0]
+      )} ${validator} (current value is ${JSON.stringify(val)})."`,
+    except: (prop, val) => `The ${prop} property must not be ${val}.`,
+    minimum: (prop, val, validator) =>
+      `The minimum value of the ${prop} must be ${validator} (current value is ${JSON.stringify(
+        val
+      )}).`,
+    maximum: (prop, val, validator) =>
+      `The maximum value of the ${prop} must be ${validator} (current value is ${JSON.stringify(
+        val
+      )}).`,
+    pattern: (prop, val, validator) =>
+      `The ${prop} value (${val}) does not match the ${validator} pattern.`,
+    maxItems: (prop, val, validator) =>
+      `The ${prop} property must not contain more than ${validator} items (currently contains ${val.length} items).`,
+    minItems: (prop, val, validator) =>
+      `The ${prop} property must contain at least ${validator} items (currently contains ${val.length} items).`,
+    divisibleBy: (prop, val, validator) =>
+      `The ${prop} property is not divisible by ${validator} (current value is ${JSON.stringify(
+        val
+      )}).`,
+    uniqueItems: (prop) => `All items in the ${prop} property must be unique.`
+  }
+};
+
+class JsonSchemaLegacy extends JsonSchemaValidator {
+  validateSchema() {
```

Comment:
Overriding `JsonSchemaValidator.validateSchema()` to use TV4 to validate both V3 and V4 (copying previous implementation, refactored).